modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/util/VelocityMigrationUtil.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check.util;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.util.Stack;
+
+/**
+ * @author Nícolas Moura
+ */
+public class VelocityMigrationUtil {
+
+	public static boolean isVelocityStatement(String line, String statement) {
+		int previousCharIndex = line.indexOf(statement) - 1;
+
+		if ((line.indexOf(statement) == 0) ||
+			((line.charAt(previousCharIndex) != CharPool.LESS_THAN) &&
+			 (line.charAt(previousCharIndex) != CharPool.SLASH))) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	public static String removeFirstParenthesis(String line, String statement) {
+		char nextChar = line.charAt(
+			line.indexOf(statement) + statement.length());
+
+		if (nextChar == CharPool.OPEN_PARENTHESIS) {
+			return StringUtil.replaceFirst(
+				line, CharPool.OPEN_PARENTHESIS, CharPool.SPACE);
+		}
+
+		return StringUtil.removeFirst(line, StringPool.OPEN_PARENTHESIS);
+	}
```

Comment:
Hi @kevhlee, I believe this method can be simplified like this:

	public static String removeFirstParenthesis(String line) {
		char previousChar = line.charAt(
			line.indexOf(CharPool.OPEN_PARENTHESIS) - 1);

		if (previousChar != CharPool.SPACE) {
			return StringUtil.replaceFirst(
				line, CharPool.OPEN_PARENTHESIS, CharPool.SPACE);
		}

		return StringUtil.removeFirst(line, StringPool.OPEN_PARENTHESIS);
	}

What you think?

Reply:
This is necessary because a declaration of arguments can be like that:

`#foreach(xxxxx`

And only removing will break the syntax, because the argument will be concatenated with the statement declaration.